scripts/resolve_container_path.py

```diff
@@ -19,24 +29,46 @@
 
 def resolve_container_path(container_path):
     container_path = os.path.realpath(container_path)
+    mountinfo_path = '/proc/self/mountinfo'
     try:
-        with open('/proc/self/cgroup') as f:
-            # Entries in /proc/self/cgroup look like this (note the nesting):
-            # 11:name=systemd:/docker/82c1bd2…23b5bcf/docker/6547bce…60ca5a7
-            prefix, container_id = next(f).strip().split(':')[2].split('/')[-2:]
+        with open(mountinfo_path) as f:
+            log.info('Found %s', mountinfo_path)
+            # Entries of interest in /proc/self/mountinfo look as follows:
+            # 752 744 259:2 /docker/containers/dc61d9…CONTAINER_ID…/<variable> …
+            expected_prefix = '/docker/containers'
+            log.info('cgroup v2 prefix is %s', expected_prefix)
+            container_ids = set()
+            for line in f.readlines():
```

Comment:
```suggestion
            for line in f:
```

Like I said, the file object is already an iterable.